uwrt_mars_rover_hw/drivers/roboteq_driver/src/CanopenInterface.cpp

```diff
@@ -82,8 +82,7 @@ const std::unordered_map<RuntimeQuery, uint16_t> CanopenInterface::QUERY_CANOPEN
 CanopenInterface::CanopenInterface(canid_t roboteq_can_id, const std::string& ifname)
     : roboteq_can_id_(roboteq_can_id) {
   if ((socket_handle_ = socket(PF_CAN, SOCK_RAW, CAN_RAW)) < 0) {
-    printf("Error while opening socket");
-    throw - 1;
+    throw StringException("Error while opening socket");
```

Comment:
i think what I said before was wrong and you should actually be throwing `std::runtime_error`s